Overhaul powa-web documentation (#191)
powa-team/powa: docs/components/powa-web/installation.rst

.. _powa-web-manual-installation:

Installation
************

Introduction
------------

> PoWA-web is the dedicated user interface of the PoWA project.  It gathers the

Reviewer: There’s a double space before “it”.

Author: yes it's on purpose, and I usually do that kind of everywhere.

I got used to it in upstream postgres sources, and after some time I did appreciate it as I think it makes the source more readable.

> PoWA-web is the dedicated user interface of the PoWA project.  It gathers the
> data on a wanted time interval from the repository server and produces graphs,
> grids and various optimisation suggestions.

Reviewer 2: ```suggestion
PoWA-web is the dedicated user interface of the PoWA project.  It produces graphs,
grids and various optimization suggestions using the data from the repository
server.
```

Author: thanks, I retweaked it a bit more.

Prerequisites
-------------

* python >= 3.6
* psycopg2
* tornado >= 2.0

Installation
------------

The recommended way to install PoWA-archivist is to use the packaged version
available in the PGDG repositories for GNU/Linux distributions
based on Debian/Ubuntu.

For other platforms, or if you need a different version from the one provided
by the PGDG repositories, you can either use

* use the `provided container images
  <https://hub.docker.com/r/powateam/powa-web>`_,
* install it from `pypi <https://pypi.org/project/powa-web/>`_
* install it manually.

We only document the package and manual installation here, as the other
methods don't having anything specific.

Installation from Debian/Ubuntu PGDG repository
***********************************************

Please refer to the `APT PGDG repository documentation

> <https://apt.postgresql.org>` for the initial setup (which should already be
> done if you have PostgreSQL installed), and simply install

Reviewer 2: ```suggestion
<https://apt.postgresql.org>` for the initial setup (which might already be
done as this is a convenient Debian PostgreSQL repository), and simply install
```

Maybe add a a footnote to https://wiki.postgresql.org/wiki/Apt/FAQ#What.27s_the_difference_between_the_PostgreSQL_packages_provided_by_Debian.2FUbuntu_and_by_the_PGDG_apt_repository.3F ?

Author: This part is still in the context of this documentation, so if you installed powa-archivist following the PGDG repository documentation and then install powa-web using the same method then yes the initial setup should have been done.

``powa-web``.  For instance:

.. code-block:: bash

  sudo apt-get install powa-web

> 

Reviewer 2: ```suggestion

Then, jump to the configuration chapter.
```

It would be cool to add internal referential links…

Author: You mean a link to literally the next line?

Reviewer 2: No, the next line is the "Manuel installation" method chapter, this later ending with the same sentence where the "jump" is really tiny over there :)

Author: oh the "Then, jump to the configuration chapter." is a new one, I was looking at the existing one.  Sorry the github interface is so confusing and I'm not familiar with the suggested change thing.

Author: I added the "then jump" sentence with a link to the configuration section.

    * `psycopg2 <https://www.psycopg.org/>`_
    * `tornado >= 2.0 <https://www.tornadoweb.org/>`_

.. tabs::

  .. code-tab:: bash RHEL / Rocky

    # Enable the EPEL repository
    sudo dnf install -y epel-release
    crb enable

    # install the dependencies
    sudo dnf install -y python3 python3-psycopg2 python3-tornado

  .. code-tab:: bash Debian / Ubuntu

    sudo apt install -y python3 python3-psycopg2 python3-tornado

  .. code-tab:: bash Archlinux

    pacman -S python python-psycopg2 python-tornado

Then, download the latest release on `pypi
<https://pypi.org/project/powa-web/>`_,  uncompress it, and copy the sample
configuration file:

.. parsed-literal::

  wget |powa_web_download_link|
  tar -zxvf powa-web-|powa_web_release|.tar.gz
  cd powa-web-|powa_web_release|
  cp ./powa-web.conf-dist ./powa-web.conf
  ./powa-web

Then, jump on the next section to configure powa-web.

Configuration
*************

The powa-web configuration is stored as a simple python file.
Powa-web will search its config as either of these files, in this order:

* /etc/powa-web.conf
* ~/.config/powa-web.conf
* ~/.powa-web.conf
* ./powa-web.conf

> You'll then be noticed of the address and port on which the UI is available.

Reviewer 2: Maybe this is a log message after starting powa-web?

```suggestion
After starting powa-web, you'll find the address and port on which the UI is available in log messages.
```

Author: also tweaked it a bit and moved to the deployment part.

> The default is 0.0.0.0:8888, as indicated in this message:

Reviewer 2: ```suggestion
The default is 0.0.0.0:8888, as indicated in this example:
```

> * [I 161105 20:27:39 powa-web:12] Starting powa-web on 0.0.0.0:8888

Author: I had the same idea when looking at it.  I didn't notice it as first as this part has been there for quite some time.

I reworded it and moved it.  It also reminds me that I added support for native SSL connections but it's not documented yet, I will take care of that once this PR is merged.

> * [I 161105 20:27:39 powa-web:12] Starting powa-web on 0.0.0.0:8888

Author: actually I meant the deployment section, as this is where the `powa-web` script is mentioned.

The following options are required:

servers (dict):

>   A dictionary mapping server names to connection information.

Reviewer 2: ```suggestion
  A dictionary mapping PostgreSQL repository servers names to their connection informations.
```

Author: I tweaked that a bit.

The terminology in the documentation is "PoWA repository server", which implies that it's a postgres server too.

Note also that in english "information" is an uncountable noun so it's not used in a plural form.

.. _powa-web-manual-installation:

Installation
************

Introduction
------------

PoWA-web is the dedicated user interface of the PoWA project.  It produces
graphs, grids and suggest various kind of optimisations based on the data
stored on the PoWA repository server.

Prerequisites
-------------

* python >= 3.6
* psycopg2
* tornado >= 2.0

Installation
------------

The recommended way to install **powa-web** is to use the packaged version
available in the PGDG repositories for GNU/Linux distributions
based on Debian/Ubuntu.

For other platforms, or if you need a different version from the one provided

> by the PGDG repositories, you can either use

Reviewer: by the PGDG repositories, you can either:

Author: fixed

* use the `provided container images
  <https://hub.docker.com/r/powateam/powa-web>`_,

> * install it from `pypi <https://pypi.org/project/powa-web/>`_

Reviewer: ``` * install it from `pypi <https://pypi.org/project/powa-web/>`_,```

Author: added missing trailing comma

    # Enable the EPEL repository
    sudo dnf install -y epel-release
    crb enable

    # install the dependencies
    sudo dnf install -y python3 python3-psycopg2 python3-tornado

  .. code-tab:: bash Debian / Ubuntu

    sudo apt install -y python3 python3-psycopg2 python3-tornado

  .. code-tab:: bash Archlinux

    pacman -S python python-psycopg2 python-tornado

Then, download the latest release on `pypi
<https://pypi.org/project/powa-web/>`_,  uncompress it, and copy the sample
configuration file:

.. parsed-literal::

  wget |powa_web_download_link|
  tar -zxvf powa-web-|powa_web_release|.tar.gz
  cd powa-web-|powa_web_release|
  cp ./powa-web.conf-dist ./powa-web.conf
  ./powa-web

Then, jump on the :ref:`next section<powa-web-config>` to configure powa-web.

.. _powa-web-config:

Configuration
*************

The powa-web configuration is stored as a simple python file.
Powa-web will search its configuration in either of these files, in this order:

* /etc/powa-web.conf
* ~/.config/powa-web.conf
* ~/.powa-web.conf
* ./powa-web.conf

You'll then be notified of the address and port on which the UI is available.
The default is 0.0.0.0:8888, as indicated in this message:

> * [I 161105 20:27:39 powa-web:12] Starting powa-web on 0.0.0.0:8888

Reviewer: ```
.. code-block::

    [I 161105 20:27:39 powa-web:12] Starting powa-web on https://0.0.0.0:8888
```

Author: thanks, fixed
